src/healthCheck/HealthCheckServer.js

```diff
@@ -0,0 +1,111 @@
+
```

Comment:
Do you plan on spinning up a new server for every component just for the health-end points?

I am feeling that this shouldn't be part of the common library.

The health-end points for the handler components currently on have a single health-end point, but I foresee the possibility of additional operations being added on (e.g. flushing caches, viewing some in-memory state, etc).

@rmothilal let me your thoughts here.

Reply:
I agree with all the comments above, but would like more clarity on the support for additional operations comment. I believe we should have a chat with @lewisdaly around how you see this being implemented.

Reply:
So the server is an optional component, which is meant to replace the `healthcheck-server` package that we are already using in the central-event-processor and email-notifier.

So if a service has its own api, we can simply subclass the BaseHealthCheck class and add the default handler to it, but if there is no other api server being run (such as for central-event-processor, email-notifier, and central-settlement), we can use the HealthCheckServer defined here.

Reply:
No. Spinning up the server is completely optional, and only for services that don't already have their own server running (eg. `email-notifier`). 

I thought this would be a good way to replace the functionality of the the `healthcheck-server` that we are using in multiple places without adding too much complexity and need to duplicate code especially for the smaller services.